components/post_view/post_info/post_info.test.tsx

```diff
@@ -135,7 +135,7 @@ describe('components/post_view/PostInfo', () => {
         const requiredPropsWithRemovePost = {...requiredProps, actions, enableEmojiPicker: true};
 
         const wrapper = shallow(<PostInfo {...requiredPropsWithRemovePost}/>);
```

Comment:
As a tip, `enzyme` actually provides a type for the output of `shallow`, called `ShallowWrapper`.
It's a generic type of the form `ShallowWrapper<{propsType}, {stateType}, {componentType}>`

So you can write it like so, and it will avoid you have to case `instance()` later on.
```suggestion
        const wrapper: ShallowWrapper<any, any, PostInfo> = shallow(<PostInfo {...requiredPropsWithRemovePost}/>);
```

Reply:
I faced some issues which I am unable to solve: related to type mismatch. The errors are related to some fields: `reply_count` and `metadata`. Please help to solve..

Reply:
Can you post the actual error from the TS compiler?

Reply:
Here's one of the errors when I do this change @devinbinnie :
```
Type '{ post: { channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }; ... 12 more ...; shouldShowDotMenu: boolean; }' is not assignable to type 'Readonly<Props>'.
  Types of property 'post' are incompatible.
    Type '{ channel_id: string; create_at: number; delete_at: number; edit_at: number; hashtags: string; id: string; is_pinned: boolean; message: string; original_id: string; parent_id: string; pending_post_id: string; ... 4 more ...; user_id: string; }' is missing the following properties from type 'Post': reply_count, metadatats(2322)
```

Reply:
This should be solved by using the `getPostMock` function mentioned below, as it looks like those two fields are just missing from the mock.